employee_class.py

```diff
@@ -10,109 +13,136 @@ def __init__(self, name, surname, age, employment_date, day_of_leaving, phone_nu
-            return int(selection)
-        else:
-            print("Please enter number between 1 and 6")
+        while True:
+            try:
+                selection = int(input('Enter your choice here (1, 2, 3, 4, 5 or 6): '))
+                if 0 <= int(selection) < 7:
+                    return selection
+                else:
+                    print("Invalid format. Please enter number between 1 and 6")
+            except ValueError:
+                print("Invalid format. Please enter number between 1 and 6")
 
     @staticmethod
     def selection_name():
         while True:
-            name = str(input("Name: "))
-            if not name:
-                print("Name cannot be blank")
-                continue
-            elif name.isalpha():
-                return name
-            else:
-                print("Name cannot contain numbers")
+            try:
+                x = Employee._validate_name(str(input("Name: ")))
+                if x:
+                    return x
+            except ValueError:
+                print(f"Value error! Please enter valid name!")
 
     @staticmethod
     def selection_surname():
         while True:
-            surname = input("Surname: ")
-            if not surname:
-                print("Surname cannot be blank")
-                continue
-            elif surname.isalpha():
-                return surname
-            else:
-                print("Surname cannot contain numbers")
+            try:
+                x = Employee._validate_name(str(input("Surname: ")))
+                if x:
+                    return x
+            except ValueError:
+                print(f"Value error! Please enter valid surname!")
 
     @staticmethod
     def selection_age():
         while True:
-            age = input("Age: ")
-            if not age:
-                print("Age cannot be blank")
-                continue
-            elif age.isdigit() and 18 < int(age) < 100:
-                return age
-            else:
-                print("Age must be a number between 18 and 100")
+            try:
+                age = int(input("Age: "))
+                if 18 < int(age) < 100:
+                    return age
+                else:
+                    print("Invalid format! Age must be a number between 18 and 100")
+            except ValueError:
+                print("Invalid format! Age must be a number between 18 and 100")
 
     @staticmethod
     def selection_employment_date():
         while True:
-            employment_date = input('Employment date YYYY MM DD: ')
-            if not employment_date:
-                print("Employment date cannot be blank")
-            elif not employment_date.isalpha():
-                return employment_date
-            else:
-                print("Employment date must be a number")
+            try:
+                x = Employee._validate_date(int(input("Employment date. Year: ")), int(input("Month: ")),
```

Comment:
o kodėl neprašai 'enter date in yyyy-mm-dd format'
ir tada tiesiogiai parsinti datą?

Reply:
done